x-pack/solutions/observability/plugins/observability_onboarding/public/application/quickstart_flows/otel_logs/use_encoded_api_key.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { HttpSetup, IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import type { ObservabilityOnboardingAppServices } from '../../..';
+
+interface OnboardingApiKeyResponse {
+  apiKeyEncoded: string;
+}
+interface APMApiKeyResponse {
+  agentKey: { name: string; encoded: string };
+}
+
+export function useEncodedApiKey() {
+  const [encodedApiKey, setEncodedApiKey] = useState<string | null>(null);
+  const [error, setError] = useState<IHttpFetchError<ResponseErrorBody> | null>(null);
+
+  const {
+    services: {
+      http,
+      context: { isServerless },
+    },
+  } = useKibana<ObservabilityOnboardingAppServices>();
+
+  useEffect(() => {
+    requestEncodedApiKey(isServerless, http)
+      .then((apiKey) => {
+        setEncodedApiKey(apiKey);
+      })
+      .catch((err) => {
+        setError(err);
+      });
+  }, [http, isServerless]);
+
+  return { encodedApiKey, error };
+}
+
+/**
+ * Managed OTel collector for now APM API keys.
+ * See https://github.com/elastic/kibana/issues/208035#issuecomment-2654482081
+ */
+function requestEncodedApiKey(isServerless: boolean, http: HttpSetup): Promise<string> {
+  let requestPromise: Promise<string>;
+
+  if (isServerless) {
+    const timestamp = new Date().toISOString();
+
+    requestPromise = http
+      .post<APMApiKeyResponse>('/api/apm/agent_keys', {
```

Comment:
The `POST /api/apm/agent_keys` endpoint requires the `apm` and `apm_settings_write` [feature privileges](https://github.com/elastic/kibana/blob/a81e6925563c0cf9417a7c045c8e797784d8fbb4/x-pack/solutions/observability/plugins/apm/server/routes/agent_keys/route.ts#L87) that are not technically needed for these flows. The API keys created also have [metadata for apm](https://github.com/elastic/kibana/blob/a81e6925563c0cf9417a7c045c8e797784d8fbb4/x-pack/solutions/observability/plugins/apm/server/routes/agent_keys/create_agent_key.ts#L86).

Also, since it's going to be decoupled from APM in the future it might be better to use the security APIs directly (`security.createApiKey`).